src/OpalCompiler-Core/RBMethodNode.extension.st

```diff
@@ -38,12 +38,12 @@ RBMethodNode >> ensureCachedArgumentNames [
 RBMethodNode class >> errorMethodNode: selector errorMessage: messageText [	
 	| message |
 	message := RBMessageNode 
-		receiver: RBSelfNode new
+		receiver: (RBVariableNode named: self)
```

Comment:
and here is a bug due to 'self' duplications.

Reply:
fixed